### PR TITLE
ci: init markdown-lint and resolve linter issues

### PR DESCRIPTION
**Pull request by heronimus**
Closes: WORLD-983

## Overview

Add markdown & MDX linter 

## Brief Changelog

- https://github.com/marketplace/actions/markdown-lint
- fix markdown linter issues
- customize markdown linter config:
  - increase `MD013/line-length` default line length from 80 (too short), to 250 (median "looks good" line length in our docs)
  - add `MD033/no-inline-html` necessary allowed HTML element 	
- use `eslint` + `eslint-plugin-mdx` plugin for *.mdx files linter
  - disable `eslint: no-unused-expressions` rules to exclude Mintlify rest/openapi definition

## Testing and Verifying

- CI passed
- MD Lint error annotation detected on changed code:
image
image

- MDX Lint error (using eslint-mdx plugin)
image

### CODE REVIEW

**Comment by smsunarto** [comment]
Is this ready for review?

**Comment by heronimus** [comment]
It's now ready, also with [world-cli](https://github.com/Argus-Labs/world-cli/pull/50) and [SGT](https://github.com/Argus-Labs/starter-game-template/pull/39)

**Review by technicallyty — APPROVED** [review]
nice

File: .github/workflows/lint-markdown.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: Lint
+
+on:
```
Comment (smsunarto) [inline .github/workflows/lint-markdown.yml:3]: - Add .mdx linter (https://github.com/mdx-js/eslint-mdx)
- Only trigger when there are changes to `.md/.mdx` file

**Review by smsunarto — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-markdown.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      - "**.mdx"
+  push:
+    branches:
+      - main
+    paths:
+      - "**.md"
+      - "**.mdx"
+
+jobs:
+  md-lint:
```
Comment (smsunarto) [inline .github/workflows/lint-markdown.yml:17]: convention for jobs name is `[workflow name]-[jobs name]` this should be `lint-md`.

Would appreciate if you can note this down so I don't have to nit/fix it everytime

**Comment by smsunarto** [comment]
test

**Comment by heronimus** [comment]
^^ PR is ready for re-review

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": ["plugin:mdx/recommended"],
+  // lint code blocks true
```
Comment (smsunarto) [inline .eslintrc:3]:
```suggestion
```

**Review by smsunarto — APPROVED** [review]

**Pull request merged**